Assignment/intlist.c

```diff
@@ -0,0 +1,129 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ------------------------------------------------------------------
+// intlist.c
+// Program for Linked List functions.
+// ------------------------------------------------------------------------------------------------
+
+#include <malloc.h>
+#include "intlist.h"
+
+LinkedList* Create () {
+   LinkedList* list = (LinkedList*)malloc (sizeof (LinkedList)); // Allocate memory for the list structure
+   if (list == NULL) return NULL;
+   list->head = NULL; // Initialize the head pointer to NULL, Indicating an empty list
+   return list;
+}
+
+int Add (LinkedList* list, int val) {
+   Node* newNode = (Node*)malloc (sizeof (Node)); // Allocate memory for the new node
+   if (newNode == NULL) return ERROR_MEMORY_ALLOCATION;
+   newNode->data = val;
+   newNode->next = NULL;
+   if (list->head == NULL) list->head = newNode;
+   else {
+      Node* current = list->head;
+      while (current->next != NULL) {
+         current = current->next;
+      }
+      current->next = newNode;
+   }
+   return SUCCESS;
+}
+
+int Insert (LinkedList* list, int index, int val) {
+   if (index < 0) return ERROR_INVALID_INDEX;
+   Node* newNode = (Node*)malloc (sizeof (Node));
+   if (newNode == NULL) return ERROR_MEMORY_ALLOCATION;
+   newNode->data = val;
+   if (index == 0) {
+      newNode->next = list->head;
+      list->head = newNode; // Insert at the head of the list
+   }
+   else {
+      Node* current = list->head;
+      int i = 0;
+      while (current != NULL && i < index - 1) {
+         current = current->next;
+         i++;
+      }
+      if (current == NULL) {
+         free (newNode);
+         return ERROR_OUT_OF_BOUNDS;
+      }
+      newNode->next = current->next;
+      current->next = newNode;
+   }
+   return SUCCESS;
+}
+
+int RemoveAt (LinkedList* list, int index) {
+   if (index < 0) return ERROR_INVALID_INDEX;
+   Node* current = list->head;
+   Node* prev = NULL;
+   int i = 0;
+   while (current != NULL && i < index) {
+      prev = current;

```

Comment:
You can use a for loop here. It is better. The loop variable gets initialised, incremented and also discarded within the loop.